misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperExecutor.java

```diff
@@ -0,0 +1,83 @@
+package org.ballerinalang.datamapper;
+
+import com.google.gson.JsonArray;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.command.CommandUtil;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.ExecuteCommandContext;
+import org.ballerinalang.langserver.commons.command.CommandArgument;
+import org.ballerinalang.langserver.commons.command.LSCommandExecutorException;
+import org.ballerinalang.langserver.commons.command.spi.LSCommandExecutor;
+import org.ballerinalang.langserver.exception.UserErrorException;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.VersionedTextDocumentIdentifier;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Represents the command executor for data mapper code action.
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.command.spi.LSCommandExecutor")
+public class AIDataMapperExecutor implements LSCommandExecutor {
+
+    public static final String COMMAND = "GENERATE_MAPPING";
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param context
+     */
+    @Override
+    public Object execute(ExecuteCommandContext context) throws LSCommandExecutorException {
+        String uri = null;
+        Range range = null;
+        JsonArray parameters = null;
+        for (CommandArgument arg : context.getArguments()) {
+            switch (arg.key()) {
+                case CommandConstants.ARG_KEY_DOC_URI:
+                    uri = arg.valueAs(String.class);
+                    break;
+                case CommandConstants.ARG_KEY_NODE_RANGE:
+                    range = arg.valueAs(Range.class);
+                    break;
+                case "JsonVal":
```

Comment:
Since you are using "JsonVal" in multiple places, can you define it as a constant and refer that.

Reply:
`JsonVal` is no logger in the code.